packages/components/src/dropdown-menu-v2-ariakit/index.tsx

```diff
@@ -0,0 +1,317 @@
+			store={ dropdownMenuContext?.store }
+		>
+			{ prefix && (
+				<Styled.ItemPrefixWrapper>{ prefix }</Styled.ItemPrefixWrapper>
+			) }
+			{ children }
+			{ suffix && (
+				<Styled.ItemSuffixWrapper>{ suffix }</Styled.ItemSuffixWrapper>
+			) }
+		</Styled.DropdownMenuItem>
+	);
+} );
+
+export const DropdownMenuCheckboxItem = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< DropdownMenuCheckboxItemProps, 'div', false >
+>( function DropdownMenuCheckboxItem(
+	{ suffix, children, hideOnClick = false, ...props },
+	ref
+) {
+	const dropdownMenuContext = useContext( DropdownMenuContext );
+
+	return (
+		<Styled.DropdownMenuCheckboxItem
+			ref={ ref }
+			{ ...props }
+			hideOnClick={ hideOnClick }
+			store={ dropdownMenuContext?.store }
+		>
+			<Ariakit.MenuItemCheck
+				store={ dropdownMenuContext?.store }
+				render={ <Styled.ItemPrefixWrapper /> }
+			>
+				<Icon icon={ check } size={ 24 } />
+			</Ariakit.MenuItemCheck>
+
+			{ children }
+			{ suffix && (
+				<Styled.ItemSuffixWrapper>{ suffix }</Styled.ItemSuffixWrapper>
+			) }
+		</Styled.DropdownMenuCheckboxItem>
+	);
+} );
+
+export const DropdownMenuRadioItem = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< DropdownMenuRadioItemProps, 'div', false >
+>( function DropdownMenuRadioItem(
+	{ suffix, children, hideOnClick = false, ...props },
+	ref
+) {
+	const dropdownMenuContext = useContext( DropdownMenuContext );
+
+	return (
+		<Styled.DropdownMenuRadioItem
+			ref={ ref }
+			{ ...props }
+			hideOnClick={ hideOnClick }
+			store={ dropdownMenuContext?.store }
+		>
+			<Ariakit.MenuItemCheck
+				store={ dropdownMenuContext?.store }
+				render={ <Styled.ItemPrefixWrapper /> }
+			>
+				<SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+					<Circle
+						cx={ 12 }
+						cy={ 12 }
+						r={ 3 }
+						fill="currentColor"
+					></Circle>
+				</SVG>
+			</Ariakit.MenuItemCheck>
+			{ children }
+			{ suffix }
+		</Styled.DropdownMenuRadioItem>
+	);
+} );
+
+export const DropdownMenuGroup = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< DropdownMenuGroupProps, 'div', false >
+>( function DropdownMenuGroup( props, ref ) {
+	const dropdownMenuContext = useContext( DropdownMenuContext );
+	return (
+		<Styled.DropdownMenuGroup
+			ref={ ref }
+			{ ...props }
+			store={ dropdownMenuContext?.store }
+		/>
+	);
+} );
+
+export const DropdownMenuGroupLabel = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< DropdownMenuGroupLabelProps, 'div', false >
+>( function DropdownMenuGroupLabel( props, ref ) {
+	const dropdownMenuContext = useContext( DropdownMenuContext );
+	return (
+		<Styled.DropdownMenuGroupLabel
+			ref={ ref }
+			{ ...props }
+			store={ dropdownMenuContext?.store }
+		/>
+	);
+} );
+
+const UnconnectedDropdownMenu = (
+	props: WordPressComponentProps< DropdownMenuProps, 'div', false >,
+	ref: React.ForwardedRef< HTMLDivElement >
+) => {
+	const {
+		// Store props
+		open,
+		defaultOpen,
```

Comment:
Would it be better to explicitly default this to `false`, so it's always a boolean and never `undefined`?

Reply:
Sounds good! https://github.com/WordPress/gutenberg/pull/54939/commits/30d982c1a9b6af9c2e8b811b05c3b28d67761b3a